### PR TITLE
Design refresh of the gallery example

### PR DESCRIPTION
**Pull request by FloVanGH**
* Visual refresh of the gallery example
* Add new Navigation 
* Improve layout
* Fix layout behavior of material `GroupBox`

image
image

### CODE REVIEW

File: examples/gallery/ui/gallery.slint [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+App := Window {
+  min-width: 800px;
+    min-height: 800px;
+    title: "Slint Gallery";
+    icon: @image-url("../../../logo/slint-logo-small-light-128x128.png");
+
+    HorizontalLayout {  
+        side-bar := SideBar {  
+            title: "Slint gallery";
```
Comment (tronical) [inline examples/gallery/ui/gallery.slint:18]: Suggestion to use title casing for the title and add "Widgets" to make it clear that this is about widgets :)
 
```suggestion
            title: "Slint Widgets Gallery";
```

**Comment by ogoffart** [comment]
How does it look in the native style?

**Comment by ogoffart** [comment]
> How does it look in the native style?

Like this on linux:

image

Not so great :-/

**Comment by ogoffart** [comment]
There are a bunch of url out there that points to examples/gallery/gallery.slint ,  If we move the slint files in ui/,   they will need to be fixed.

**Comment by FloVanGH** [comment]
I moved the `gallery.slint` back to the old position.

**Comment by FloVanGH** [comment]
Ok I see, I will check it also on my linux vm so it looks nice there ether.  Native macos style looks great.

**Comment by FloVanGH** [comment]
image

**Comment by ogoffart** [comment]
I think the problem on my screenshot bight be because of the scale factor and the layout doesn't seem to do its job.

File: examples/gallery/gallery.slint [hunk 2]
```diff
@@ -1,170 +1,46 @@
...
+import { AboutPage, ControlsPage, ListViewPage, TextEditPage } from "ui/pages/pages.slint";
+import { GallerySettings } from "ui/gallery_settings.slint";
+import { Navigator } from "ui/navigator.slint";
+import { SideBar } from "ui/side_bar.slint";
 
 App := Window {
-    preferred-width: 500px;
-    preferred-height: 600px;
+  min-width: 760px;
+    min-height: 680px;
```
Comment (ogoffart) [inline examples/gallery/gallery.slint:12]: ideally, the minimum size should be computed based on the minimum size of the contents by the layout.

File: examples/gallery/ui/navigator.slint [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+export Navigator := Rectangle {
+    property<int> current-item: 0;
+
+    height: 100%;
```
[the remark this answers is not in the document]
Comment (FloVanGH) [reply examples/gallery/ui/navigator.slint:7]: What is not necessary height: 100% ?

File: examples/gallery/gallery.slint (hunk 2 above)
Comment (FloVanGH) [reply examples/gallery/gallery.slint:12]: Yes I have to check a better way there.

Comment (FloVanGH) [reply examples/gallery/gallery.slint]: Should be fixed. Can you check it on Linux?

**Comment by ogoffart** [comment]
Another thing: the current gallery can be controlled almost entirely with the keyboard.  But that nice side widget currently does not handle tabs very well.

**Comment by FloVanGH** [comment]
Thank you for the hint. That is a good point. I will add focus management to the SideBar.

**Comment by FloVanGH** [comment]
The SideBar works now similar to the TabWidget

**Comment by ogoffart** [comment]
Current state on linux



The window starts a bit too small.  
(Also I liked the scrolling between pane in the first prototype)

File: examples/gallery/gallery.slint [hunk 4]
```diff
@@ -1,171 +1,26 @@
...
+import { AboutPage, ControlsPage, ListViewPage, TextEditPage } from "ui/pages/pages.slint";
+import { GallerySettings } from "ui/gallery_settings.slint";
+import { SideBar } from "ui/side_bar.slint";
 
 App := Window {
-    preferred-width: 500px;
-    preferred-height: 600px;
-    title: "Slint Gallery";
+    preferred-width: 700px;
+    min-height: 500px;
```
Comment (ogoffart) [inline examples/gallery/gallery.slint:11]:
```suggestion
    preferred-height: 500px;
```

**Pull request merged**